src/geometry/CCPACSGenericSystem.cpp

```diff
@@ -40,7 +40,7 @@ namespace tigl
 // Constructor
 CCPACSGenericSystem::CCPACSGenericSystem(CCPACSGenericSystems* parent, CTiglUIDManager* uidMgr)
 : generated::CPACSGenericSystem(parent, uidMgr)
-, CTiglRelativelyPositionedComponent(NULL, &m_transformation, &m_symmetry)
+, CTiglRelativelyPositionedComponent(static_cast<std::string*>(nullptr), &m_transformation, &m_symmetry)
```

Comment:
It looks like you are explicitly casting so that the new constructor of `CTiglRelativelyPositionedComponent` without the optional `parentUid` is called, correct? Out of curiosity: Why is this needed? Does this imply that from now on generic systems cannot be parentless root components anymore? While this seems reasonable, I don't understand why it would need to be enforced.

Reply:
I had to add a new Constructor to CTiglRelativelyPositionedComponent. With this addition, there are now two constructors, that accept a pointer as a first argument:
```cpp
  CTiglRelativelyPositionedComponent(std::string* ....);
  CTiglRelativelyPositionedComponent(boost::optional<std::string>* ....);
```
The compiler does not know, which to chose when just providing NULL so you have to make the pointer type explicit.

And yes, a generic systems does not seem to have a ```parentUID``` element, that's why we have to add the ```nullptr``` here.